test/e2e/apimachinery/apiserver_identity.go

```diff
@@ -37,7 +37,7 @@ import (
 	admissionapi "k8s.io/pod-security-admission/api"
 )
 
-func getControlPlaneHostname(node *v1.Node) (string, error) {
+func getControlPlaneHostname(ctx context.Context, node *v1.Node) (string, error) {
```

Comment:
I suggest to merge this commit in a follow up PR, so we have some time to observe the behavior of the CI.

Reply:
"This commit" = "[e2e: use context from Ginkgo](https://github.com/kubernetes/kubernetes/pull/112923/commits/8838fc4314822ae079132486bb0c056c8ff59e3a)"?

But then the rest of the changes are pretty harmless (I know, famous last words...). How long would you want to test those before doing the real change?

Reply:
some soak time, that is better than doing a full revert later or start bisecting ... it seems that if we do this today we have to merge the other tomorrow https://www.kubernetes.dev/resources/release/

Reply:
I've split out the "easy" parts into [a separate PR](https://github.com/kubernetes/kubernetes/pull/113884) and rebased this one here on top of it so that the complex commit is at the end.

We can first merge that and then this one (they won't conflict), or go directly for this one.

Reply:
defer to @spiffxp